tools/phone-data-migration/src/PhoneDataMigration/Lib.hs

```diff
@@ -0,0 +1,126 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2024 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module PhoneDataMigration.Lib where
+
+import Cassandra as C
+import Data.Conduit
+import qualified Data.Conduit.Combinators as Conduit
+import Data.Id
+import Imports
+import Options.Applicative
+import PhoneDataMigration.Types
+import System.Logger.Class (MonadLogger)
+import qualified System.Logger.Class as Log
+import System.Logger.Message ((.=), (~~))
+import Wire.API.User (Email)
+
+pageSize :: Int32
+pageSize = 1000
+
+dropExcludedPhonesTable :: (MonadClient m) => m ()
+dropExcludedPhonesTable = retry x1 . write cql $ paramsP LocalQuorum () pageSize
+  where
+    cql :: PrepQuery W () ()
+    cql = "TRUNCATE TABLE excluded_phones"
+
+getUsers :: (MonadClient m) => ConduitM () [User] m ()
+getUsers =
+  paginateC cql (paramsP LocalQuorum () pageSize) x5
+    .| Conduit.map (fmap (\(uid, phone, email, ssoId) -> User uid (phone >>= parsePhone) email (isJust ssoId)))
+  where
+    cql :: PrepQuery R () (UserId, Maybe Text, Maybe Email, Maybe Text)
+    cql = "SELECT id, phone, email, sso_id FROM user"
+
+dropPhoneFromUser :: (MonadClient m) => UserId -> m ()
+dropPhoneFromUser =
+  retry x5
+    . write cql
+    . params LocalQuorum
+    . Identity
+  where
+    cql :: PrepQuery W (Identity UserId) ()
+    cql = "UPDATE user SET phone = null WHERE id = ?"
```

Comment:
why don't you drop the column?  then you don't need to go through the contents and delete fields individually.

Reply:
(also, it *should* be deleted :)